sound/soc/sof/intel/cnl.c

```diff
@@ -188,6 +188,29 @@ static int cnl_ipc_send_msg(struct snd_sof_dev *sdev,
 	return 0;
 }
 
+int cnl_dai_fixup(struct snd_sof_dev *sdev,
+		  struct snd_soc_dai_link *link,
+		  struct sof_ipc_dai_config *config,
+		  int direction)
+{
+	int ret = 0;
+
+	switch (config->type) {
+	case SOF_DAI_INTEL_SSP:
+	case SOF_DAI_INTEL_DMIC:
+		break;
+	case SOF_DAI_INTEL_HDA:
+		ret = hda_dai_fixup(sdev, link, config, direction);
+		break;
+	default:
+		dev_err(sdev->dev, "error: invalid DAI type %d\n", config->type);
+		ret = -EINVAL;
+		break;
+	}
+
+	return ret;
+}
+
```

Comment:
Meh. copy/paste code isn't really accepted in Linux circles.

Reply:
yes, I will refine it later